config/fcos/v1_6_exp/validate.go

```diff
@@ -51,11 +52,24 @@ func (conf Config) Validate(c path.ContextPath) (r report.Report) {
 func (d BootDevice) Validate(c path.ContextPath) (r report.Report) {
 	if d.Layout != nil {
 		switch *d.Layout {
-		case "aarch64", "ppc64le", "x86_64":
+		case "aarch64", "ppc64le", "x86_64", "s390x-eckd", "s390x-virt", "s390x-zfcp":
 		default:
 			r.AddOnError(c.Append("layout"), common.ErrUnknownBootDeviceLayout)
 		}
 	}
+	//Validate s390x layout device specific luks.device-s390x.
+	//s390x layout does not support Mirror.
+	if d.Layout != nil {
+		s390xLayout = []string{"s390x-zfcp", "s390x-eckd"}
+		for _, zlayout := range s390xLayout {
+			switch {
```

Comment:
Two if's might more more appropriate here?

Reply:
Is this ok?
```
if d.Layout != nil && (*d.Layout == "s390x-zfcp" || *d.Layout == "s390x-eckd") {
		if util.NilOrEmpty(d.Luks.Device) {
                            r.AddOnError(c.Append(zlayout), common.ErrNoLuksBootDevice)
                 }
                 if len(d.Mirror.Devices) > 0 {
                          r.AddOnError(c.Append(zlayout), common.ErrMirrorNotSupport)
                  }
}
```

Reply:
``` 
if d.Layout != nil {
		switch *d.Layout {
		case "s390x-eckd", "s390x-zfcp":
			if util.NilOrEmpty(d.Luks.Device) {
				r.AddOnError(c.Append(*d.Layout), common.ErrNoLuksBootDevice)
			}
			if len(d.Mirror.Devices) > 0 {
				r.AddOnError(c.Append(*d.Layout), common.ErrMirrorNotSupport)
			}
			if *d.Layout == "s390x-zfcp" && util.NotEmpty(d.Luks.Device) && !sdRe.MatchString(*d.Luks.Device) {
				r.AddOnError(c.Append(*d.Layout), common.ErrNoLuksBootDevice) ```